From c4c043940be1d5add1ac24a5ccac146c7b430f9a Mon Sep 17 00:00:00 2001
From: yaozenhu <yaozenhu@realtek.com>
Date: Mon, 6 Mar 2017 01:57:29 +0000
Subject: [PATCH 11/11] [fix bug] RDK-200 [root cause] h264parse find out
 resolution will be chaged chadbut don't send pop frame event, so cause
 gst-omx wait all buffer return [solution] execute resolution process only by
 rtk-omx, ignore h264parse inform resolution change event [reviewer] Paul

Change-Id: I230974f2b5baba7d507c745873129b872211e229
---
 gst/videoparsers/gsth264parse.c | 11 ++++++++---
 gst/videoparsers/gsth264parse.h |  1 +
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/gst/videoparsers/gsth264parse.c b/gst/videoparsers/gsth264parse.c
index 05b6b75..6dcea6f 100644
--- a/gst/videoparsers/gsth264parse.c
+++ b/gst/videoparsers/gsth264parse.c
@@ -215,6 +215,7 @@ gst_h264_parse_reset_stream_info (GstH264Parse * h264parse)
   h264parse->multiview_mode = GST_VIDEO_MULTIVIEW_MODE_NONE;
   h264parse->multiview_flags = GST_VIDEO_MULTIVIEW_FLAGS_NONE;
   h264parse->first_in_bundle = TRUE;
+  h264parse->first_change = FALSE;
 
   h264parse->align = GST_H264_PARSE_ALIGN_NONE;
   h264parse->format = GST_H264_PARSE_FORMAT_NONE;
@@ -1734,9 +1735,11 @@ gst_h264_parse_update_src_caps (GstH264Parse * h264parse, GstCaps * caps)
             h264parse->height != crop_height)) {
       GST_INFO_OBJECT (h264parse, "resolution changed %dx%d",
           crop_width, crop_height);
-      h264parse->width = crop_width;
-      h264parse->height = crop_height;
-      modified = TRUE;
+      if(h264parse->first_change == FALSE){
+	      h264parse->width = crop_width;
+	      h264parse->height = crop_height;
+	      modified = TRUE;
+      }
     }
 
     /* 0/1 is set as the default in the codec parser, we will set
@@ -1905,6 +1908,8 @@ gst_h264_parse_update_src_caps (GstH264Parse * h264parse, GstCaps * caps)
   gst_caps_unref (sink_caps);
   if (buf)
     gst_buffer_unref (buf);
+
+  h264parse->first_change = TRUE;
 }
 
 static void
diff --git a/gst/videoparsers/gsth264parse.h b/gst/videoparsers/gsth264parse.h
index 58d818c..436e0ab 100644
--- a/gst/videoparsers/gsth264parse.h
+++ b/gst/videoparsers/gsth264parse.h
@@ -122,6 +122,7 @@ struct _GstH264Parse
 
   GstClockTime pending_key_unit_ts;
   GstEvent *force_key_unit_event;
+  gboolean first_change;
 
   /* Stereo / multiview info */
   GstVideoMultiviewMode multiview_mode;
-- 
2.7.4

